### PR TITLE
Blink title when new messages come in

### LINKED ISSUE #40: Favicon or page title "reply waiting" indicator would be great.
It would be awesome if the favicon or page title could flash when there's a reply waiting for the user, so when they're multitasking they can know there's something ready for them.

### PR DESCRIPTION
**Pull request by lemieux**
This should fix #40 .

The title of the page will alternate between the original title and "{User} said...". This should draw attention to the tab. 

@dannytranlx @Mario54 @mspensieri

### CODE REVIEW

File: src/js/components/message-indicator.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import DocumentTitle from 'react-document-title';
+
+const BLINKING_INTERVAL = 1500;
+
+export class MessageIndicatorComponent extends Component {
+    state = {
+        initialDocumentTitle: global.document ? document.title : '',
```
Comment (mspensieri) [inline src/js/components/message-indicator.jsx]: Should this value be refreshed at some point? If the user switches pages or the page title changes at some point, our value will be stale

Comment (lemieux) [inline src/js/components/message-indicator.jsx]: Good point, but I wonder how we can effectively track that. I mean, what if the page change occurs while the message is shown... I guess the user's app will update the title when changing url...?

Comment (mspensieri) [inline src/js/components/message-indicator.jsx]: I think before every change we should first check if the title is what we expect, something like this:

```
onTick() {
  if(title === "{name} said"){
    title = this.initialTitle;
  }else if(title == this.initialTitle){
    title = '{name} said';
  }else{
    this.initialTitle = title;
    title = '{name} said';
  }
}
```

Comment (lemieux) [inline src/js/components/message-indicator.jsx]: yeah good idea, fix incoming

File: src/js/components/message-indicator.jsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    };
+
+    blinkTitle() {
+        if (!this.blinkInterval) {
+            const fn = () => {
+                const {unreadCount, messages, messageIndicatorTitle} = this.props;
+                const {currentTitle} = this.state;
+                let {initialDocumentTitle} = this.state;
+                
+                if (document.title !== initialDocumentTitle && !document.title.endsWith(messageIndicatorTitle.replace('{name}', ''))) {
```
Comment (lemieux) [inline src/js/components/message-indicator.jsx]: @mspensieri like this?

Comment (mspensieri) [inline src/js/components/message-indicator.jsx]: Yep. Why `endsWith` though?

Comment (lemieux) [inline src/js/components/message-indicator.jsx]: fastest way to check if it's using the text in the strings list without having to go through the conversation to fetch the last appMaker's name.

Comment (mspensieri) [inline src/js/components/message-indicator.jsx]: Don't think that check works if the guy customized his string though

```
"User Mike replied".endsWith("User {name} replied".replace('{name}', ''))

=> false
```

Comment (lemieux) [inline src/js/components/message-indicator.jsx]: true :(

Comment (lemieux) [inline src/js/components/message-indicator.jsx]: I'll add a reverse loop to start from the end of the conversation and get that name

Comment (mspensieri) [inline src/js/components/message-indicator.jsx]: What if we just stored the last set "smooch" title in the state?

File: README.md [hunk 3]
```diff
@@ -99,7 +99,12 @@ var skPromise = Smooch.init({
         settingsInputPlaceholder: 'Your email address',
         settingsSaveButtonText: 'Save',
         settingsHeaderText: 'Email Settings',
-        settingsNotificationText: 'In case we\'re slow to respond you can <a href="#" data-ui-settings-link>leave us your email</a>.'
+        settingsNotificationText: 'In case we\'re slow to respond you can <a href data-ui-settings-link>leave us your email</a>.',
+        actionPaymentError: 'An error occurred while processing the card. <br> Please try again or use a different card.',
+        actionPaymentCompleted: 'Payment Completed',
+        messageError: 'An error occured while sending your message. Please try again.',
+        invalidFileError: 'Sorry, but only images are supported currently. Please choose a file with a supported extension (jpg, jpeg, png, gif, or bmp).',
+        messageIndicatorTitle: '{name} said...'
```
Comment (jpjoyal) [inline README.md]: Let's go with "(1) New message" & (2) "New messages" as name is will generate a 🤔  face most of the time for the user. Its very likely that its the first time you see this name

Comment (lemieux) [inline README.md]: done.

File: src/js/components/widget.jsx [hunk 4]
```diff
@@ -10,16 +10,17 @@ import { Settings } from 'components/settings';
 import { Notification } from 'components/notification';
 import { ErrorNotification } from 'components/error-notification';
 import { ChatInput } from 'components/chat-input';
+import { MessageIndicator } from 'components/message-indicator';
```
Comment (Mario54) [inline src/js/components/widget.jsx]: nit, call this `NewMessageIndicator`?

**Comment by Mario54** [comment]
👍

**Comment by mspensieri** [comment]
:+1:

**Pull request merged**